### PR TITLE
EWPP-435: Update publication teaser view display.

### PR DESCRIPTION
**Pull request by ademarco**

### CODE REVIEW

**Review by drishu — DISMISSED** [review]

File: modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php [hunk 1]
```diff
@@ -128,3 +128,23 @@ function oe_theme_content_publication_post_update_00007() {
...
+
+  // Take over teaser view display, regardless if it already exists or not.
+  $view_display = EntityViewDisplay::load($display_values['id']);
+  if ($view_display) {
+    $display = $storage->updateFromStorageRecord($view_display, $display_values);
+    $display->save();
+    return;
+  }
+
+  $display = $storage->createFromStorageRecord($display_values);
```
Comment (yenyasinn) [inline modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php:153]: Teaser view display is created in `oe_theme_content_publication_post_update_00004()`. If it was removed do we need to create it again? Looks like yes.

If yes, then `$config['_core']['default_config_hash'] = Crypt::hashBase64(serialize($config));` should be implemented prior to `$storage->createFromStorageRecord`. See oe_theme_content_news_post_update_00007().

Comment (ademarco) [reply modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php:153]: That's correct, @georgebaev you are working on it right?

Comment (georgebaev) [reply modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php:153]: I added the config_hash related functionality.

**Comment by georgebaev** [comment]
The commit at https://github.com/openeuropa/oe_theme/pull/629/commits/504bac8087d10fe7811c0e4c21f62788dda63c4e fixes the problem where the `\n` character is required in the tests. After the PR at https://github.com/openeuropa/oe_theme/pull/629 is merged, we'll have to update the tests where the `\n` character is set.

File: modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php [hunk 2]
```diff
@@ -128,3 +128,28 @@ function oe_theme_content_publication_post_update_00007() {
...
+  $display_values = $storage->read('core.entity_view_display.node.oe_publication.teaser');
+  // We are creating the config which means that we are also shipping
+  // it in the config/install folder so we want to make sure it gets the hash
+  // so Drupal treats it as a shipped config. This means that it gets exposed
+  // to be translated via the locale system as well.
+  $display_values['_core']['default_config_hash'] = Crypt::hashBase64(serialize($display_values));
+  $storage = \Drupal::entityTypeManager()->getStorage('entity_view_display');
+
+  // Take over teaser view display, regardless if it already exists or not.
+  $view_display = EntityViewDisplay::load($display_values['id']);
```
Comment (yenyasinn) [inline modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php]:
```suggestion
  $view_display = $storage->load($display_values['id']);
```

Comment (georgebaev) [reply modules/oe_theme_content_publication/oe_theme_content_publication.post_update.php]: This is fixed.

File: templates/content/node--oe-publication--teaser.html.twig [hunk 3]
```diff
@@ -4,12 +4,16 @@
...
 #}
+{% set image_url = content.oe_publication_thumbnail|field_value|render %}
 {{ pattern('list_item', {
-  'variant': 'thumbnail_primary',
+  'variant': 'thumbnail_secondary',
   'url': url,
   'title': content.extra_field_oe_theme_helper_short_title_with_fallback|field_value,
   'meta': [
-    content.oe_publication_date|field_value
+    content.oe_publication_type|field_value|safe_join(', '),
```
Comment (yenyasinn) [inline templates/content/node--oe-publication--teaser.html.twig:13]: On the mockup pipe is used as separator between Resource type items.
Screenshot 2020-11-27 at 12 39 54

From spec - "This is the Primary Meta (fields are merged using "|")" - might be also it should be used for Author items too.

**Review by imanoleguskiza — DISMISSED** [review]

**Review by imanoleguskiza — DISMISSED** [review]

**Review by yenyasinn — DISMISSED** [review]

File: templates/content/node--oe-publication--teaser.html.twig [hunk 4]
```diff
@@ -4,12 +4,16 @@
...
 {{ pattern('list_item', {
-  'variant': 'thumbnail_primary',
+  'variant': 'thumbnail_secondary',
   'url': url,
   'title': content.extra_field_oe_theme_helper_short_title_with_fallback|field_value,
   'meta': [
-    content.oe_publication_date|field_value
+    content.oe_publication_type|field_value|safe_join(' | '),
+    content.oe_publication_date|field_value,
+    content.oe_author|field_value|safe_join(' | ')
```
Comment (ademarco) [inline templates/content/node--oe-publication--teaser.html.twig]: This was correct as it was before. With @kugta  we had agreed to use commas to separate multiple values of the same type, and pipes for things of different types.

File: templates/content/node--oe-publication--teaser.html.twig (hunk 3 above)
Comment (ademarco) [reply templates/content/node--oe-publication--teaser.html.twig:13]: @kugta  we should re-align requirements based on the above.

**Review by yenyasinn — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**